### PR TITLE
web: fix memory leak warning

### PR DESCRIPTION
**Pull request by jazzdan**
Following the advice here https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component/

### CODE REVIEW

File: web/src/AppController.ts [hunk 1]
```diff
@@ -32,6 +33,14 @@ class AppController {
     this.loadCount = 0
   }
 
+  componentDidMount() {
```
Comment (nicks) [inline web/src/AppController.ts]: where do you expect these methods to be called? Note that AppController isn't a react component.

Comment (jazzdan) [reply web/src/AppController.ts]: D'oh...

/me drinks more coffee

File: web/src/HUD.tsx [hunk 2]
```diff
@@ -82,10 +85,14 @@ class HUD extends Component<HudProps, HudState> {
 
   componentDidMount() {
     this.controller.createNewSocket()
+    this._isMounted = true
   }
 
   componentWillUnmount() {
-    this.controller.dispose()
+    if (this._isMounted) {
+      this.controller.dispose()
```
Comment (nicks) [inline web/src/HUD.tsx]: sorry, i still don't understand this pr. why should the component need to be mounted to dispose the controller?

**Comment by nicks** [comment]
can you give more context on what problem you're trying to fix? i think you're very confused about something.

**Comment by jazzdan** [comment]
I'm going to close this because I can't reproduce the error message.

**Pull request closed without merging**